src/eventgenerator/generator/evaluator.go

```diff
@@ -102,10 +113,9 @@ func (e *Evaluator) retrieveAppMetrics(trigger *model.Trigger) ([]*model.AppMetr
 }
 func (e *Evaluator) sendTriggerAlarm(trigger *model.Trigger) {
 	url := e.scalingEngineUrl
-	b := new(bytes.Buffer)
-	json.NewEncoder(b).Encode(trigger)
-	path := strings.Replace(PathScale, "{appid}", trigger.AppId, 1)
-	resp, respErr := e.httpClient.Post(url+path, "", b)
+	jsonBytes, _ := json.Marshal(trigger)
```

Comment:
The issue is that json maps need to have keys that are strings. Go 1.7 has fixed this so let us upgrade to Go 1.7 and now you should be able to check the error.